subprojects/core-api/src/main/java/org/gradle/api/artifacts/DependencySubstitutions.java

```diff
@@ -71,6 +72,25 @@
      */
     ComponentSelector project(String path);
 
+    /**
+     * Transforms the supplied selector into a specific variant selector.
+     * @param selector the origin selector
+     * @param detailsAction the variant selection details configuration
+     *
+     * @since 6.6
+     */
+    @Incubating
+    ComponentSelector variant(ComponentSelector selector, Action<? super VariantSelectionDetails> detailsAction);
```

Comment:
I am not sure about the best place in the javadoc, but we should clearly state that the attributes specified here need only be the ones _different_ from those of the resolution context.
Maye we can simply reference the documentation for adding attributes on dependency declarations. The applicability is the same.